build.gradle.kts

```diff
@@ -15,10 +16,55 @@ tasks.register<Exec>("installGitHooks") {
 val initialTaskNames: List<String> = project.gradle.startParameter.taskNames
 project.gradle.startParameter.setTaskNames(initialTaskNames + listOf("installGitHooks"))
 
-tasks.register("detektAll").configure {
+val detektAllTask = tasks.register("detektAll") {
     gradle.includedBuilds
         .filterNot { it.name == "build-logic-settings" }
         .forEach { includedBuild ->
             dependsOn(includedBuild.task(":detektAll"))
         }
 }
+
+// Register lifecycle tasks in this umbrella build.
+// A user/CI usually only needs these.
+val checkAll = tasks.register("checkAll") {
+    group = taskGroup
+    description = "Run all tests and create code coverage report"
+
+    dependsOn(
+        gradle.includedBuilds
+            .filter { !it.name.startsWith("build-logic") }
```

Comment:
Can we hide this duplicated knowledge about "build-logic"?
Maybe reusable function or make fake empty tasks inside build-logic.

Reply:
i made tasks in build-logic projects